war/src/main/java/io/lumeer/engine/rest/CollectionService.java

```diff
@@ -576,11 +580,141 @@ public void dropAttributeConstraint(final @PathParam("collectionCode") String co
          throw new UserCollectionNotFoundException(ErrorMessageBuilder.userCollectionNotFoundString(collectionCode));
       }
 
-      if (!securityFacade.hasCollectionRole(projectCode, collectionCode, LumeerConst.Security.ROLE_WRITE)) {
+      if (!collectionMetadataFacade.hasRole(projectCode, collectionCode, LumeerConst.Security.ROLE_WRITE)) {
          throw new UnauthorizedAccessException();
       }
 
       collectionMetadataFacade.dropAttributeConstraint(collectionCode, attributeName, constraintConfiguration);
    }
 
+   @GET
+   @Path("/{collectionCode}/permissions")
+   public Map<String, List<Permission>> getPermissions(final @PathParam("collectionCode") String collectionCode) throws CollectionNotFoundException, UnauthorizedAccessException {
+      if (collectionCode == null) {
+         throw new BadRequestException();
+      }
+
+      if (!collectionFacade.hasCollection(collectionCode)) {
+         throw new UserCollectionNotFoundException(ErrorMessageBuilder.userCollectionNotFoundString(collectionCode));
+      }
+
+      if (!collectionMetadataFacade.hasRole(projectCode, collectionCode, LumeerConst.Security.ROLE_READ)) {
```

Comment:
Users with `read` role should not be able to see collection permissions. You need to have `manage` role in order to do so.